nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -133,7 +105,10 @@ public void initialize(String inputName, Map<String, String> options) {
       throw new IllegalStateException("Parameter 'warehouse' not set, Nessie can't store data.");
     }
     final String requestedRef = options.get(removePrefix.apply(NessieConfigConstants.CONF_NESSIE_REF));
-    this.reference = loadReference(requestedRef, null);
+    NessieApiV1 api = createNessieClientBuilder(options.get(NessieConfigConstants.CONF_NESSIE_CLIENT_BUILDER_IMPL))
```

Comment:
Maybe move this (lines 107..111) into a static builder-ish method in NessieIcebergClient?

Reply:
I have another cleanup PR in the pipeline and will adjust this part of the code accordingly there